src/include/migraphx/op/scatternd.hpp

```diff
@@ -0,0 +1,81 @@
+#ifndef MIGRAPHX_GUARD_OPERATORS_SCATTERND_HPP
+#define MIGRAPHX_GUARD_OPERATORS_SCATTERND_HPP
+
+#include <array>
+#include <migraphx/check_shapes.hpp>
+#include <migraphx/stringutils.hpp>
+#include <migraphx/streamutils.hpp>
+#include <migraphx/shape_for_each.hpp>
+#include <migraphx/par_for.hpp>
+#include <migraphx/config.hpp>
+#include <migraphx/value.hpp>
+#include <migraphx/op/normalize_attribute.hpp>
+#include <cmath>
+#include <utility>
+#include <type_traits>
+
+namespace migraphx {
+inline namespace MIGRAPHX_INLINE_NS {
+namespace op {
+
+struct scatternd
+{
+    std::string reduction = "none";
+
+    template <class Self, class F>
+    static auto reflect(Self& self, F f)
+    {
+        return pack(f(self.reduction, "reduction"));
+    }
+
+    std::string name() const { return "scatternd"; }
+
+    shape compute_shape(std::vector<shape> inputs) const
+    {
+        check_shapes{inputs, *this}.has(3).standard();
```

Comment:
We should not require `standard` shape for ref implementation. If we use the iterators for tensor_view(instead of pointers), then it should work for non-standard shapes.